.github/workflows/publish-to-npm.yml

```diff
@@ -103,3 +103,41 @@ jobs:
       - run: |
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc
           npm dist-tag add @salesforce/sfdx-scanner@${{ github.event.release.tag_name || inputs.tag }} latest-beta
+  # Step 5: Create a Pull Request for merging `main` into `dev`
+  create-main2dev-pull-request:
+    needs: promote-to-latest
+    runs-on: macos-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      # Check out `main`
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      # Create a new branch based on `main`, so that merge conflicts can be manually resolved if need be.
+      - run: |
+          NEW_VERSION=$(jq -r ".version" package.json)
+          git checkout -b m2d/v$NEW_VERSION
+          git push --set-upstream origin m2d/v$NEW_VERSION
+      # Create a Pull Request from the new branch into `dev`.
+      - run: |
+          NEW_VERSION=$(jq -r ".version" package.json)
+          # For whatever reason, the version of 'echo' on GHAs doesn't process backspace by default.
+          # The non-POSIX-standard -e flag causes it to do that.
+          echo -e "This branch and PR were automatically created following the successful release of v$NEW_VERSION.\n\
+          It must be MERGED into dev, NOT SQUASHED OR REBASED. Squashing or rebasing this branch onto dev can cause potentially irreconcilable merge conflicts later.\n\
+          As an additional safeguard and reminder, the title of this PR MUST include the word 'merging' in the description portion of the PR title, e.g., 'Main2Dev @W-XXXXXXX@ Merging main to dev after vX.Y.Z'.\n\
+          If there are conflicts between dev and this branch, you should do the following locally:\n\
+          - $ git checkout dev
+          - $ git pull
+          - $ git fetch --all
+          - $ git checkout m2d/v$NEW_VERSION
+          - $ git pull origin dev --no-rebase # You MUST include this flag, or someone's day will be ruined.
+          - Resolve the merge conflicts manually. When in doubt, ask the code's author for help.
+          - $ git commit
+          - $ git push" > body.txt
+          # Create the pull request.
+          gh pr create -B dev -H m2d/v$NEW_VERSION --title "Filler title. Read description and rename." -F body.txt
```

Comment:
`gh pr create` accepts a `--body` flag, but using that for multi-line strings is weird and inconvenient. So I opted instead to write the intended body to a file and use the `-F` flag.